schema/shared/src/main/scala/zio/blocks/schema/Patch.scala

```diff
@@ -14,21 +14,54 @@ package zio.blocks.schema
  *
  * patch3(Person("Jane", 25)) // Person("John", 30)
  * }}}
+ *
+ * Patches can also be created by diffing two values:
+ * {{{
+ * val person1 = Person("Jane", 25)
+ * val person2 = Person("John", 30)
+ * val patch = Schema[Person].diff(person1, person2)
+ * patch(person1) // Person("John", 30)
+ * }}}
  */
-final case class Patch[S](ops: Vector[Patch.Pair[S, ?]], source: Schema[S]) {
+final case class Patch[S](
+  ops: Vector[Patch.Pair[S, ?]],
+  source: Schema[S],
+  dynamicPatch: Option[DynamicPatch] = None
+) {
   import Patch._
 
-  def ++(that: Patch[S]): Patch[S] = Patch(this.ops ++ that.ops, this.source)
+  def ++(that: Patch[S]): Patch[S] = {
+    val mergedOps = this.ops ++ that.ops
+    val mergedDynamic = (this.dynamicPatch, that.dynamicPatch) match {
+      case (Some(d1), Some(d2)) => Some(d1 ++ d2)
+      case (Some(d1), None) => Some(d1)
+      case (None, Some(d2)) => Some(d2)
+      case (None, None) => None
+    }
+    Patch(mergedOps, this.source, mergedDynamic)
+  }
 
-  def apply(s: S): S =
-    ops.foldLeft[S](s) { (s, single) =>
+  def apply(s: S): S = {
+    // First apply typed ops
```

Comment:
Very poorly defined. There should be a single dynamic patch that is applied to the source value, rather than a weird combination of "typed ops" and untyped patch.